### PR TITLE
Adds support for cpu utilization on mac with resolution of under 100ms

### PR DESCRIPTION
**Pull request by ribalba**
Closes https://github.com/green-coding-solutions/green-metrics-tool/issues/330
Closes https://github.com/green-coding-solutions/green-metrics-tool/pull/344

I choose to make a new pr as it seemed cleaner.

### CODE REVIEW

**Comment by ribalba** [comment]
Why are you renaming it to mach? We don't use mach anywhere. It is mac everywhere?

File: metric_providers/cpu/utilization/mac/system/source.c [hunk 1]
```diff
@@ -35,7 +35,7 @@ void loop_utilization(unsigned int msleep_time) {
             }
 
             gettimeofday(&now, NULL);
-            printf("%ld%06i %i\n", now.tv_sec, now.tv_usec, (int)(ut_total * 100 / numCPUsU));
+            printf("%ld%06i %i\n", now.tv_sec, now.tv_usec, (int)( (ut_total / (float)numCPUsU)*100*100) );
```
Comment (ribalba) [inline metric_providers/cpu/utilization/mac/system/source.c]: Good catch. I forgot that we multiply by 100 again.

**Comment by ArneTR** [comment]
Works great, really nice commit!

I patched only that the utiliatzion was in percent and not in ratio. Other than that some name changes and display additons.

TY!

@dan-mm Please use this provider for your mac tests

**Comment by ArneTR** [comment]
> Why are you renaming it to mach? We don't use mach anywhere. It is mac everywhere?

Yeah sorry, I did not clarify. Our [naming scheme](https://docs.green-coding.io/docs/measuring/metric-providers/metric-providers-overview/#naming-convention) specifies that this part of the string should declare the source. Mac is not really the source, but the mach syscall library is. Similar to the procfs as the vfs as source in the other CPU utilization provider. Hope that makes it more clearer.

**Pull request merged**